tinygrad/tensor.py

```diff
@@ -2363,7 +2363,8 @@ def sub(self, x:Union[Tensor, ConstType], reverse=False) -> Tensor:
     print(t.sub(Tensor([[2.0], [3.5]])).numpy())
     ```
     """
-    return F.Sub.apply(*self._broadcasted(x, reverse))
+    a, b = self._broadcasted(x, reverse)
```

Comment:
don't need this `_broadcasted`, it's handed by add

Reply:
Changing to `return -(self + (-x)) if reverse else self + (-x)` increases the kernel count of a few tests by 1. For example for `test_reduce_multiple_paths_midreduce`, a new kernel with NEG will be added 
```
kernel 3
  0 ━┳ STORE MemBuffer(idx=0, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(), strides=(), offset=0, mask=None, contiguous=True),)))
  1  ┗━┳ NEG 
  2    ┗━━ LOAD MemBuffer(idx=1, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(), strides=(), offset=0, mask=None, contiguous=True),)))
kernel 4
  0 ━┳ STORE MemBuffer(idx=0, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(1, 1), strides=(0, 0), offset=0, mask=None, contiguous=True),)))
  1  ┗━┳ MAX (0, 1)
  2    ┗━┳ ADD 
  3      ┣━━ LOAD MemBuffer(idx=1, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(4, 4), strides=(4, 1), offset=0, mask=None, contiguous=True),)))
  4      ┗━━ LOAD MemBuffer(idx=2, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(4, 4), strides=(0, 0), offset=0, mask=None, contiguous=False),)))
```
while previously when `_broadcasted` was used, 
```
kernel 3
  0 ━┳ STORE MemBuffer(idx=0, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(1, 1), strides=(0, 0), offset=0, mask=None, contiguous=True),)))
  1  ┗━┳ MAX (0, 1)
  2    ┗━┳ ADD 
  3      ┣━━ LOAD MemBuffer(idx=1, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(4, 4), strides=(4, 1), offset=0, mask=None, contiguous=True),)))
  4      ┗━┳ NEG 
  5        ┗━━ LOAD MemBuffer(idx=2, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(4, 4), strides=(0, 0), offset=0, mask=None, contiguous=False),)))
```

@chenyuxyz I'd prefer to keep `_broadcasted` here to simplify the kernels. WDYT?

Reply:
i see, because it expands the const, sounds good